src/messages/mod.rs

```diff
@@ -331,7 +331,7 @@ impl Welcome {
     }
 
     /// Get a reference to the ciphersuite in this Welcome message.
-    pub(crate) fn get_secrets_ref(&self) -> &[EncryptedGroupSecrets] {
```

Comment:
The `EncryptedGroupSecrets` contain the hashes of the `KeyPackages`, which we need to determine which `KeyPackageBundle` we need to decrypt and process it. Again, we either need to expose it or expose a function that allows us to handle this case in some other way.